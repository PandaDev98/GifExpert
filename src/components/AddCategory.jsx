import { useState } from "react"
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')

    const onSubmit = (event) => {
        event.preventDefault();
        const trimValue = inputValue.trim()

        if (trimValue.length <= 1) { return }
        onNewCategory(trimValue)
        setInputValue('');
    }
    return (
        <form aria-label="form" action="" onSubmit={onSubmit}>
            <input
                id="categoryInput"
                type="text"
                placeholder="Buscar Gifs de..."
                value={inputValue}
                onChange={event => { setInputValue(event.target.value) }}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}