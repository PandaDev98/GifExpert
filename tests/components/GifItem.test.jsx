import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Test on GifItem', () => {

    const title = 'Saitama';
    const url = 'https://test.com/'

    test('should match with snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    });

    test('should show the image with the rigth url and alt', () => {
        render(<GifItem title={title} url={url} />)
        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('should show the title onthe app', () => {
        render(<GifItem title={title} url={url} />)        
        expect( screen.getByText(title)).toBeTruthy();
        

    });
});