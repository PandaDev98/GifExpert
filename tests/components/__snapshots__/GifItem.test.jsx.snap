// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifItem should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://test.com/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
